Frontend/HsTypeChecker.hs

```diff
@@ -519,6 +495,161 @@ rnTyVarsToFcTypes = map rnTyVarToFcType
 rnTmVarToFcTerm :: RnTmVar -> FcTerm
 rnTmVarToFcTerm = FcTmVar . rnTmVarToFcTmVar
 
+-- * Pattern Match Elaboration
+-- ----------------------------------------------------
+
+-- | Suprised all this doesn't exist yet? Is it needed?
```

Comment:
I would suggest either 
- instantiating the SubstVar class, so you can just use the substVar method.
- or creating a new class for this, since I realize that this code is a bit more specific (since it only 'renames' variables).
- or finally reusing the renamer somehow (since replacing variables with variables is exactly what it does), but I don't really see an easy way of reusing this code either tbh.